File: artifacts/api-server-py/app/stripe_client.py
"""Stripe client factory — port of stripeClient.ts.

Verified against stripe==15.2.1 in this session: `StripeClient` exists, but
its top-level resource shortcuts (`.products`, `.checkout`, ...) are
deprecated in favor of the explicit `.v1` namespace (`.v1.products`,
`.v1.checkout`, ...). All resources also expose `*_async` methods, which is
what makes `StripeClient` worth using over the classic
`stripe.api_key = "..."` global-mutation style in an async FastAPI app.
"""

import stripe

from app.config import STRIPE_SECRET_KEY, STRIPE_WEBHOOK_SECRET


def get_stripe_client() -> stripe.StripeClient:
    if not STRIPE_SECRET_KEY:
        raise RuntimeError(
            "Stripe integration not connected. Set the STRIPE_SECRET_KEY "
            "environment variable to enable payments."
        )
    return stripe.StripeClient(STRIPE_SECRET_KEY)


def get_stripe_webhook_secret() -> str | None:
    return STRIPE_WEBHOOK_SECRET

File: _deprecated/api-server/src/stripeClient.ts
import Stripe from "stripe";

function getStripeCredentials(): { secretKey: string; webhookSecret?: string } {
  const secretKey = process.env.STRIPE_SECRET_KEY;
  if (!secretKey) {
    throw new Error(
      "Stripe integration not connected. Set the STRIPE_SECRET_KEY environment variable to enable payments.",
    );
  }
  return {
    secretKey,
    webhookSecret: process.env.STRIPE_WEBHOOK_SECRET,
  };
}

export async function getUncachableStripeClient(): Promise<Stripe> {
  const { secretKey } = getStripeCredentials();
  return new Stripe(secretKey);
}

export function getStripeWebhookSecret(): string | undefined {
  return process.env.STRIPE_WEBHOOK_SECRET;
}
